cmd/root.go

```diff
@@ -14,12 +17,66 @@ var (
 		Short: "bat-go provides go based services and processes for BAT",
 	}
 	ctx = context.Background()
+
+	// variables will be overwritten at build time
+	version   string
+	commit    string
+	buildTime string
+
+	// top level config items
+	pprofEnabled      string
+	env               string
+	ratiosAccessToken string
+	ratiosService     string
 )
 
+// helper to make sure there is no errors
+func must(err error) {
+	if err != nil {
+		log.Printf("failed to initialize: %s\n", err.Error())
+		os.Exit(1)
+	}
+}
+
 // Execute - the main entrypoint for all subcommands in bat-go
 func Execute() {
+	// setup context with logging
+	var logger *zerolog.Logger
+	ctx, logger = logging.SetupLogger(ctx)
+
+	// execute the root cmd
 	if err := rootCmd.Execute(); err != nil {
-		log.Printf("error running bat-go: %s\n", err.Error())
+		logger.Error().Err(err).Msg("./bat-go command encountered an error")
 		os.Exit(1)
 	}
 }
+
+func init() {
+
+	// pprof-enabled - defaults to ""
+	rootCmd.PersistentFlags().StringVarP(&pprofEnabled, "pprof-enabled", "", "",
+		"pprof enablement")
+	must(viper.BindPFlag("pprof-enabled", rootCmd.PersistentFlags().Lookup("pprof-enabled")))
+	must(viper.BindEnv("pprof-enabled", "PPROF_ENABLED"))
+
+	// env - defaults to development
+	rootCmd.PersistentFlags().StringVarP(&env, "environment", "e", "development",
+		"the default environment")
+	must(viper.BindPFlag("environment", rootCmd.PersistentFlags().Lookup("environment")))
+	must(viper.BindEnv("environment", "ENV"))
+	// must(rootCmd.MarkPersistentFlagRequired("environment"))
```

Comment:
Will these be commented out when we deploy?

Reply:
It seems like `MarkPersistentFlagRequired` requires that the flag be set on the command line, and doesn't recognize it can be in the environment as well.

Reply:
Until we can figure out a work around, we are not going to use required, as we want to use env variables for configs as well and not have to specify on command line.

Reply:
https://github.com/brave-intl/bat-go/issues/408